pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/MutableDictionaryReader.java

```diff
@@ -186,4 +202,63 @@ public void print() {
   public boolean isEmpty() {
     return dictionaryIdBiMap.isEmpty();
   }
+
+  /**
+   * Checks if the given value lies within one of the partition ranges. If the value is not within
+   * any of the partition ranges, partitioning is dropped. The effect of that is:
+   * <ul>
+   *   <li> Subsequent values will not be checked against partition ranges. </li>
+   *   <li> Partition information will not be written out to the metadata for this column. </li>
+   * </ul>
+   *
+   * @param value Column value to check.
+   */
+  protected void checkPartition(Object value) {
```

Comment:
Partitioning is not a dictionary concern and should be placed in either `RealtimeSegmentStatsContainer` or a related class. It makes no sense design-wise to have a partition function on a dictionary, especially considering partitioning only makes sense on a segment basis.